offchain/hello-world-api/src/Signing.purs

```diff
@@ -0,0 +1,57 @@
+module Signing
+  ( getPubKey
+  , hsmSignTx
+  , getCmd
+  ) where
+
+import Contract.Prelude
+
+import Contract.Address (ByteArray)
+import Contract.Prim.ByteArray (byteArrayToHex)
+import Contract.Transaction (Ed25519Signature(..), PublicKey(..), Transaction(Transaction), TransactionWitnessSet, Vkey(..), _vkeys)
+import Ctl.Internal.Serialization (toBytes)
+import Ctl.Internal.Serialization as Serialization
+import Data.Lens (set)
+import Data.String (trim)
+import Effect.Aff (makeAff)
+import Effect.Exception (throw)
+import Node.Buffer.Class (toString)
+import Node.ChildProcess (ExecResult, defaultExecOptions, execFile)
+import Node.Encoding (Encoding(UTF8))
+import Node.Process (lookupEnv)
+import Untagged.Union (asOneOf)
+
+type Signer = ByteArray -> Aff String
+
+getCmd :: String -> Aff String
+getCmd varName = liftEffect $ do
+  lookupEnv varName >>= case _ of
+    Just cmd -> pure cmd
+    Nothing -> throw $ "expected " <> varName <> " to be set"
+
+getPubKey :: Aff PublicKey
+getPubKey = do
+  (res :: ExecResult) <- execAff "signer" [ "getPubKey" ]
+  PublicKey <<< trim <$> (liftEffect $ toString UTF8 res.stdout)
+
+hsmSignTx :: PublicKey -> Transaction -> Aff TransactionWitnessSet
+hsmSignTx = signTx cmdSigner
+
+cmdSigner :: Signer
+cmdSigner a = do
+  let args = [ "sign", byteArrayToHex a ]
+  (res :: ExecResult) <- execAff "signer" args
+  liftEffect $ trim <$> toString UTF8 res.stdout
+
+execAff :: String -> Array String -> Aff ExecResult
+execAff file args = makeAff $ \callBack -> do
```

Comment:
[execFile](https://github.com/purescript-node/purescript-node-child-process/blob/v9.0.0/src/Node/ChildProcess.purs#L311-L316) executes a command rather than a file, even though the name of the function is saying something else

Reply:
I've renamed it cmd because fair enough it does sound better but I'm sorta confused by the distinction.
Could `execFile` execute shell aliases or something? Certainly in our case the command is also a file.

Reply:
I don't know how shell aliases are resolved tbh, whether you need a shell environment or not.